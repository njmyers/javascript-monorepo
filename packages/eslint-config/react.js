module.exports = {
  parser: 'babel-eslint',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  env: {
    'shared-node-browser': true,
  },
  extends: ['plugin:react/recommended'],
  plugins: ['react', 'react-hooks', 'jsx-a11y'],
  rules: {
    'jsx-a11y/label-has-associated-control': 'error',
    'react/display-name': 'off',
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
};
