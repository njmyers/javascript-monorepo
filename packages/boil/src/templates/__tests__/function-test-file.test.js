import * as React from 'react';
import functionTestFile from '../function-test-file';

describe('it works', () => {
  test('it is a function ', () => {
    expect(typeof functionTestFile).toBe('function');
  });

  test('it runs without crashing', () => {
    expect(functionTestFile()).toMatchSnapshot();
  });
});


