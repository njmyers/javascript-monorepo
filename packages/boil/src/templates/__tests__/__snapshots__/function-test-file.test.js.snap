// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it runs without crashing 1`] = `
"import testFunction from '../test-function';

describe('it works', () => {
  test('it is a function ', () => {
    expect(typeof testFunction).toBe('function');
  });

  test('it runs without crashing', () => {
    expect(testFunction()).toMatchSnapshot();
  });
});
"
`;
